modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -389,19 +389,19 @@ private void updateLeaseBatchInternal() {
             Map<ReplicationGroupId, Boolean> toBeNegotiated = new HashMap<>();
             Map<ReplicationGroupId, Lease> renewedLeases = new HashMap<>(leasesCurrent.leaseByGroupId());
 
-            Map<ReplicationGroupId, TokenizedAssignments> currentAssignments = assignmentsTracker.assignments();
-            Set<ReplicationGroupId> currentAssignmentsReplicationGroupIds = currentAssignments.keySet();
+            Map<ReplicationGroupId, TokenizedAssignments> currentStableAssignments = assignmentsTracker.stableAssignments();
+            Set<ReplicationGroupId> currentStableAssignmentsReplicationGroupIds = currentStableAssignments.keySet();
 
             // Remove all expired leases that are no longer present in assignments.
             renewedLeases.entrySet().removeIf(e -> clockService.before(e.getValue().getExpirationTime(), now)
-                    && !currentAssignmentsReplicationGroupIds.contains(e.getKey()));
+                    && !currentStableAssignmentsReplicationGroupIds.contains(e.getKey()));
 
-            int currentAssignmentsSize = currentAssignments.size();
+            int currentStableAssignmentsSize = currentStableAssignments.size();
             int activeLeasesCount = 0;
 
-            for (Map.Entry<ReplicationGroupId, TokenizedAssignments> entry : currentAssignments.entrySet()) {
+            for (Map.Entry<ReplicationGroupId, TokenizedAssignments> entry : currentStableAssignments.entrySet()) {
```

Comment:
Why do you think all entries are in stable assignment?
As I understand it, we are using pending assessment because it might be that the stable assignment does not exist for a specific group.

Reply:
I guess we haven't the such case. On table creation we're posting in MS our first stables and then after rebalances we're only make stable switch => we always have stable assignment for a group

Reply:
Both should. Added unionization method that aggregate pendings and stables tokenized assignments maps onto map of assignments sets pairs by group id and use the pair in the cycle's body